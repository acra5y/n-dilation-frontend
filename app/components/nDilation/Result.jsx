import React from "react";
import styled from "styled-components";

import withAnimatedContentChange from "../withAnimatedContentChange";
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator";
import Matrix from "./Matrix";
import ErrorMessage from "./ErrorMessage";

const AnimatedContentTransition = styled.div`
    text-align: center;
    transition: ${props => `opacity ${props.animationTimeInSeconds}s ease-out`};
    opacity: ${props => props.opacity};
`;

export const Result = ({
    isLoading,
    dilation,
    validationError,
    runtimeError,
    opacity,
    animationTimeInSeconds,
}) => {
    return (
        <AnimatedContentTransition
            opacity={opacity}
            animationTimeInSeconds={animationTimeInSeconds}
        >
            {isLoading ? (
                <LoadingIndicator />
            ) : (
                <>
                    {dilation && <Matrix matrixInRowMajorOrder={dilation} />}
                    {validationError ||
                        (runtimeError && (
                            <ErrorMessage validationError={validationError} />
                        ))}
                </>
            )}
        </AnimatedContentTransition>
    );
};

export default withAnimatedContentChange(Result);
