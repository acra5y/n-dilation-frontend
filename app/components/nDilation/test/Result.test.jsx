import React from "react";
import { shallow } from "enzyme";

import { Result } from "../Result";

describe("Result component", () => {
    const testCases = [
        {
            props: {},
            description:
                "should render correct markup when error and dilation are undefined",
        },
        {
            props: { isLoading: true },
            description: "should render correct markup when isLoading is true",
        },
        {
            props: { dilation: [1, 2, 3, 4] },
            description:
                "should render correct markup dilation prop is defined",
        },
        {
            props: {
                dilation: [1, 2, 3, 4],
                runtimeError: true,
            },
            description:
                "should render correct markup dilation and runtimeError is true",
        },
        {
            props: { validationError: true },
            description: "should render correct markup validationError is true",
        },
    ];
    testCases.forEach(({ props, description }) => {
        it(description, () => {
            expect(shallow(<Result {...props} />)).toMatchSnapshot();
        });
    });
});
