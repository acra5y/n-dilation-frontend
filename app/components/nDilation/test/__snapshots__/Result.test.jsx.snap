// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result component should render correct markup dilation and runtimeError is true 1`] = `
<Result__AnimatedContentTransition>
  <_default
    matrixInRowMajorOrder={
      Array [
        1,
        2,
        3,
        4,
      ]
    }
  />
  <ErrorMessage />
</Result__AnimatedContentTransition>
`;

exports[`Result component should render correct markup dilation prop is defined 1`] = `
<Result__AnimatedContentTransition>
  <_default
    matrixInRowMajorOrder={
      Array [
        1,
        2,
        3,
        4,
      ]
    }
  />
</Result__AnimatedContentTransition>
`;

exports[`Result component should render correct markup validationError is true 1`] = `
<Result__AnimatedContentTransition>
  <Component />
</Result__AnimatedContentTransition>
`;

exports[`Result component should render correct markup when error and dilation are undefined 1`] = `<Result__AnimatedContentTransition />`;

exports[`Result component should render correct markup when isLoading is true 1`] = `
<Result__AnimatedContentTransition>
  <LoadingIndicator
    getRandomInt={[Function]}
  />
</Result__AnimatedContentTransition>
`;
